lib/fs.js

```diff
@@ -1617,8 +1617,15 @@ function ReadStream(path, options) {
   if (!(this instanceof ReadStream))
     return new ReadStream(path, options);
 
+  if (options === undefined)
+    options = {};
+  else if (typeof options === 'string')
+    options = { encoding: options };
+  else if (options === null || typeof options !== 'object')
+    throw new TypeError('options must be a string or an object');
+
   // a little bit bigger buffer and water marks by default
-  options = Object.create(options || {});
+  options = Object.create(options);
```

Comment:
Your concern is `Object.create(null);` throws null exception? 
`Object.create(null)` does not throw null exception, and I prevent `null` on line 1624.

If you concern about the backward compatible, we discussed here (https://github.com/nodejs/io.js/pull/1845#discussion_r31453308)